backend/site-api/src/db/cache_helpers.rs

```diff
@@ -0,0 +1,75 @@
+use lib_shared_site_api::{cache::cache::SiteMetadata, error::api_error::ApiError};
+use lib_shared_types::dto::site_api::get_current_site_dto::{
+    to_api_response, GetCurrentSiteResponse,
+};
+
+use crate::api_context::ApiContext;
+
+pub async fn get_site_from_cache_or_repo(
+    context: &ApiContext,
+    site_id: &str,
+) -> Result<GetCurrentSiteResponse, ApiError> {
+    match context.cache.get_usage(site_id).await {
+        Ok(usage) => {
+            let cache: Result<GetCurrentSiteResponse, serde_json::Error> =
+                serde_json::from_value(usage.site_data);
+
+            match cache {
+                Ok(site) => return Ok(site),
+                Err(_) => (),
+            }
+        }
+        Err(_) => (),
+    };
+
+    let site = context
+        .site_repo
+        .get_site_latest_version(site_id, true)
+        .await
+        .map_err(|e| ApiError::not_found().message(e))?;
+    Ok(to_api_response(&site, site.pages.clone()))
+}
+
+pub async fn get_metadata_from_cache_or_repo(
+    context: &ApiContext,
+    site_id: &str,
+) -> Result<SiteMetadata, ApiError> {
+    context
+        .cache
+        .get_metadata_with(site_id, async move {
+            let meta = context
+                .metadata_repo
+                .get_site_metadata(&site_id)
+                .await
+                .map_err(|e| ApiError::not_found().message(e))?;
+
+            Ok(SiteMetadata {
+                owner_id: meta.owner_id,
+                site_type: meta.site_type,
+                disabled: meta.disabled,
+            })
+        })
+        .await
+    /*
```

Comment:
I forgot to remove this code 😅, I'll do it in my next PR. This function gets metadata from the cache, or updates the cache from the database if it's missing. I think the site data cache could be implemented similarly, but it was working fine and is a little more complex, so I left it as-is for now.

One thing I'm wondering is whether it's a good idea to use `Arc` for the site_data cache, and/or separate it from the usage. The reason is that `moka` clones the data when retrieving it, so we're potentially using a lot of memory/CPU every time it's accessed. It's kind of hard to measure though!